lib/api/controller/security.js

```diff
@@ -619,36 +619,28 @@ class SecurityController extends NativeController {
    * @returns {Promise<Object>}
    */
   async getUserStrategies(request) {
-    const id = this.getId(request);
-    const promises = [];
+    const userId = this.getId(request);
+    const strategies = [];
 
     // Throws if the user doesn't exist
-    await this.ask('core:security:user:get', id);
+    await this.ask('core:security:user:get', userId);
 
-    if (this.anonymousId === id) {
-      promises.push(Bluebird.resolve([]));
-    }
-    else {
+    if (this.anonymousId !== userId) {
       const availableStrategies = global.kuzzle.pluginsManager.listStrategies();
 
       for (const strategy of availableStrategies) {
-        const existStrategy = this.getStrategyMethod(strategy, 'exists');
+        const existMethod = this.getStrategyMethod(strategy, 'exists');
 
-        promises.push(
-          existStrategy(request, id, strategy)
-            .then(exists => exists ? strategy : null));
+        if (await existMethod(request, userId, strategy)) {
```

Comment:
Here you first code was better because it was a parallel processing and now it's sequential.
`async/await` should be used unless you can/need parallel processing

Reply:
Ofc.